### PR TITLE
MAKE-322 parallelize getting process info

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: message/process_info.go [hunk 1]
```diff
@@ -84,19 +85,42 @@ func CollectProcessInfoWithChildren(pid int32) []Composer {
 // message.Composer interfaces for all processes currently running on
 // a system.
 func CollectAllProcesses() []Composer {
+	numThreads := 32
```
Comment (tychoish) [inline message/process_info.go]: shouldn't we make the numThreads be 25% of number of procs (ish) up to some reasonable max, like 32?

File: message/process_info.go [hunk 2]
```diff
@@ -84,19 +85,42 @@ func CollectProcessInfoWithChildren(pid int32) []Composer {
...
+	doneChan := make(chan bool)
+	go func() {
+		defer close(doneChan)
+		for p := range infoChan {
+			results = append(results, p)
+		}
+	}()
+	wg.Wait()
+	close(infoChan)
+	<-doneChan
```
Comment (tychoish) [inline message/process_info.go]: avoid the extra thread to collect results, and just buffer the infoChan

File: message/process_info.go (hunk 1 above)
Comment (john-m-liu) [reply message/process_info.go]: I made it the number of processes, up to 32

File: message/process_info.go (hunk 2 above)
Comment (john-m-liu) [reply message/process_info.go]: done

**Pull request merged**